backend/python_init_script/main.py

```diff
@@ -2,41 +2,186 @@
+    """
+    Reads the given file and parses it into UserData objects.
+
+    :param file: path of file containing the user data that shall be loaded
+    :type file: str
+    :return: iterable of the loaded UserData objects
+    :rtype: [UserData]
+    """
+    ending = str(file).lower().split(".")[-1]
+    user_df = pd.DataFrame()
+    if ending == "csv":
+        user_df = pd.read_csv(file, dtype=str)
+        # if there are less than 5 columns -> there might be ';' used as separators
+        if user_df.shape[1] < 5:
+            user_df = pd.read_csv(file, sep=";", dtype=str)
+    elif ending in ["xlsx", "xls"]:
+        user_df = pd.read_excel(file, dtype=str)
+    else:
+        print("Only csv and xlsx format are supported!")
+
+    for row_id, row in user_df.iterrows():
+        # if cell has no value in xlsx/csv it will result in a nan value
+        # -> replace with empty string
+        # -> split roles by ',' if there are more
+        _roles = str(row["Roles"])
+        _roles = [] if _roles == "nan" else _roles.split(",")
+        _team = str(row["Team"])
+        _team = "" if _team == "nan" else _team
+
+        yield UserData(
+            name=str(row["Name"]),
+            email=str(row["Email"]),
+            team=_team,
+            roles=_roles,
+            init_pw=str(row["InitialPassword"]),
+        )
 
 
 def init_client(
-    project_id: str = os.environ.get("APPWRITE_FUNCTION_PROJECT_ID"),
-    endpoint: str = os.environ.get("APPWRITE_ENDPOINT"),
-    api_key: str = os.environ.get("APPWRITE_API_KEY"),
+    project_id: str = os.getenv("APPWRITE_PROJECT_ID"),
+    endpoint: str = os.getenv("APPWRITE_ENDPOINT"),
+    api_key: str = os.getenv("APPWRITE_API_KEY"),
 ) -> Client:
+    """
+    Initializes the appwrite client object.
+
+    :param project_id: Project ID of the targeted appwrite project
+    :type project_id: str
+    :param endpoint: URL of the appwrite endpoint
+    :type endpoint: str
+    :param api_key: API key to access appwrite backend
+    :type api_key: str
+    :return: Initialized appwrite client object
+    :rtype: Client
+    """
     _client = Client()
     _client.set_endpoint(endpoint)
     _client.set_project(project_id)
     _client.set_key(api_key)
     return _client
 
 
-client = init_client()
-users = Users(client)
-teams = Teams(client)
+def get_or_create_team_id(teams: Teams, team_name: str) -> str:
+    """
+    Gets team id of given team name, if no team with this name is found the method creates the team.
+    If it finds more then one team of the given name, an exception is raised.
+
+    :param teams: teams object, containing the server information
+    :type teams: Teams
+    :param team_name: team name which id shall be returned
+    :type team_name: str
+    :return: ID of the team
+    :rtype: str
+    """
+    result = teams.list()
+    team_ids = [x["$id"] if x["name"] == team_name else None for x in result["teams"]]
+    team_ids = list(filter(None, team_ids))
```

Comment:
Also nice could be:
`team_ids = filterTeamIDsByName(team_name, teams.list())`

Reply:
done

Reply:
I'm sorry, I actually know the the underscore convention in Python but I forgot it :-D .